lsls/lsls.c

```diff
@@ -4,15 +4,45 @@
 /**
  * Main
  */
-int main(int argc, char **argv)
+int main(int argc, char **argv) // main takes in a argv pointer. Is argc number or args given?
 {
-  // Parse command line
+  struct dirent *pointDir; // struct that finds serial # and Directory name
+  char *directoryName; // Store directory name in a pointer
 
-  // Open directory
+  if(argc == 1) // need to check that arg[1] is set to something
+  {
+    directoryName = "."; // set dir
 
-  // Repeatly read and print entries
+  } else if(argc == 2) {
+
+    directoryName = opendir(argv[1]);
```

Comment:
The return value of `opendir` is a `DIR` type so instead we should assign `directoryName` `argv[1]` instead.